source/tutorials/learning-journey/shell-dot-nix.md

```diff
@@ -0,0 +1,159 @@
+# TITLE
+
+<!-- Include any foreward you want here -->
+
+## Overview
+
+### What will you learn?
+
+<!-- Give a brief description of what the reader will learn so that they know whether the topic interests them. -->
+- How to create reproducible shell environments
+
+### How long will it take?
+WIP
+
+<!-- Give some indication of how long it will take to complete the tutorial so that the reader knows whether to continue. -->
+
+### What will you need?
+
+<!-- List any prerequisite knowledge or tools the reader will need to complete the tutorial. -->
+- A basic understanding of the Nix language
+
+## Entering a shell with Python installed
+Suppose we wanted to enter a shell in which Python 3.10 was installed.
+The simplest possible way to accomplish this is via the `nix-shell -p` command:
+```
+$ nix-shell -p python310
+```
+
+This command works, but there's a number of inefficiences:
+- You have to type out `-p python310` every time you enter the shell.
+- It doesn't scale to an arbitrary number of packages (you would have to type out each package name each time).
+- It doesn't (ergonomically) allow you any further customization of your shell environment.
+
+A better solution is to create our shell environment from a `shell.nix` file.
+
+## A basic `shell.nix` file
+The `nix-shell` command by default looks for a file called `shell.nix` in the current directory and tries to build a shell environment by evaluating the Nix expression in this file.
+So, if you properly describe the shell environment you want in a `shell.nix` file, you can enter it with just the `nix-shell` command without any further arguments.
+No more specifying packages on the command line.
+Here's what a basic `shell.nix` looks like that installs Python 3.10 as before:
+```nix
+let
+    pkgs = import <nixpkgs> {};
+in
+    pkgs.mkShell {
+        packages = [
+          pkgs.python310
+        ];
+    }
+```
+where `mkShell` is a function that when called produces a shell environment.
+
+If you save this into a file called `shell.nix` and call `nix-shell` in the directory containing this `shell.nix` file, you'll enter a shell with Python 3.10 installed.
+
+## Adding packages
+Additional executable packages are added to the shell by adding them to the `packages` attribute.
+For example, let's say we wanted to add `curl` to our shell environment.
+The new `shell.nix` would look like this:
+```nix
+let
+    pkgs = import <nixpkgs> {};
+in
+    pkgs.mkShell {
+        packages = [
+          pkgs.python310
+          pkgs.curl  # new package
+        ];
+    }
+```
+
+TODO: go into `packages` vs. `buildInputs`
```

Comment:
```suggestion
:::{note}
`nix-shell` was originally conceived as a way to construct a shell environment containing the tools needed to *develop software*; only later was it widely used as a general way to construct temporary environments for other purposes. `mkShell` is a [wrapper around `mkDerivation`](https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-mkShell), so you could also add packages to the `buildInputs` or `nativeBuildInputs` [attributes](https://nixos.org/manual/nixpkgs/stable/#sec-using-stdenv) instead of `packages`, and get the same result.
:::
```

I'm actually inclined to move this somewhere else; this is supposed to be one of the reader's first encounters with Nix.